### ISSUE TITLE
Documentation for Python computations a little sparse

### ISSUE THREAD

**Issue by alexpghayes** [quarto-dev/quarto-web#147]
Apologies for how discombobulated the following comments are, updating as I puzzle things out.

Some things I came across today while trying to port some `.Rmd` with `python` sections to `.qmd`. I am looking at <https://quarto.org/docs/computations/python.html>

- I had the Python `jupyter` library installed but it was such an old version that it was insufficient to pass `quarto check`. Might add a min version number.
- I setup a new conda env `py310` and installed `jupyter` in it, but then I still had to set `QUARTO_PYTHON`
I ended up adding 

```
EXPORT QUARTO_PYTHON = "/home/alex/anaconda3/envs/py310/bin/python"
```

to my `.bashrc` (I'm on Ubuntu).

At first I thought that `QUARTO_PYTHON` had to be the same Python I used to execute the cells in the notebook, but I now see this is not the case and that I can execute the cells using any jupyter kernel, which I set using the `jupyter` parameter (I don't think this is documented anywhere). I can list all my available kernels with

```
jupyter kernelspec list
```

It would be nice if the VS Code/RStudio visual editors had some sort of click to choose kernel dropdown like Jupyter notebook itself does.

It might also be nice to point towards some basic resources on create Python envs, installing packages into them, and then turning those envs into Jupyter kernels.

**Comment by alexpghayes** [comment]
After I computer restart I realize that I somehow didn't set `QUARTO_PYTHON` correctly? Is there a recommended way to set `QUARTO_PYTHON`?

```
(base) alex@dncu:~$ echo $QUARTO_PYTHON
/home/alex/anaconda3/envs/py310/bin/python
(base) alex@dncu:~$ quarto check

[✓] Checking Quarto installation......OK
      Version: 0.9.257
      Path: /opt/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.8.3 (Conda)
      Path: /home/alex/anaconda3/bin/python
      Jupyter: 4.6.3
      Kernels: sklearn0.24, python3, sagemath

(|) Checking Jupyter engine render....Traceback (most recent call last):
  File "/opt/quarto/share/jupyter/jupyter.py", line 21, in <module>
    from notebook import notebook_execute, RestartKernel
  File "/opt/quarto/share/jupyter/notebook.py", line 14, in <module>
    from nbclient import NotebookClient
  File "/home/alex/anaconda3/lib/python3.8/site-packages/nbclient/__init__.py", line 6, in <module>
    from .client import NotebookClient, execute  # noqa: F401
  File "/home/alex/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 18, in <module>
    from traitlets import (
ImportError: cannot import name 'Callable' from 'traitlets' (/home/alex/anaconda3/lib/python3.8/site-packages/traitlets/__init__.py)
[✓] Checking Jupyter engine render....OK
```

I've tried adding 

```
EXPORT QUARTO_PYTHON = "/home/alex/anaconda3/envs/py310/bin/python"
``` 

to my `.bashrc` and also

```
QUARTO_PYTHON = "/home/alex/anaconda3/envs/py310/bin/python"
```

to `/etc/environment` and am having no luck with either.

**Comment by alexpghayes** [comment]
I appear to have entered some special hell. In RStudio I can render a `.qmd` with Python sections just fine (right), but interactively running the Python cells in the `.qmd` notebook seems to have some sort of environment breakage (left).

![image](https://user-images.githubusercontent.com/15078251/165802236-07cb2d95-ae25-42d1-88ae-d23fb1b8781f.png)

**Comment by alexpghayes** [comment]
Ah, just found https://quarto.org/docs/computations/jupyter-kernels.html, does not seemed to be linked to from the Python computations page.

**Comment by jjallaire** [comment]
QUARTO_PYTHON is intended as a last-ditch hammer (sort of like `RETICULATE_PYTHON`). From the CLI intended workflow is that Quarto will use the version of Python on your PATH (or if you are in a conda environment it will use that one).  So you need to activate the conda env and then call quarto.

In more recent versions of the VS Code extension we should be automatically using whatever version of Python is bound to the current workspace/session (we set QUARTO_PYTHON under the hood to the value the Python extension returns when we ask it which version of Python is active).

**Comment by alexpghayes** [comment]
Closing this issue because my comments are chaotic and hard to follow, will revisit and provide more specific suggestions for doc improvements in a couple days.
